insights/tests/client/test_client.py

```diff
@@ -24,63 +48,212 @@ def test_version():
         sys.argv = tmp
 
 
-@patch('insights.client.client.constants.sleep_time', 0)
-@patch('insights.client.client.InsightsConnection.upload_archive',
-       return_value=Mock(status_code=500))
-def test_upload_500_retry(upload_archive):
+def test_register():
+    config = InsightsConfig(register=True)
+    client = InsightsClient(config)
+    client.connection = FakeConnection()
+    client.session = True
+    assert client.register() is True
```

Comment:
Without any mocking here, we are touching the real filesystem. The tests are actually reading and **writing** files in the real _/etc_. That’s really bad. Not only in would need either root access or some special permissions setup, but more importantly it breaks out of the sandbox, doing mess in a real system.

Reply:
I think this is resolved now